src/main/java/me/shivzee/util/Message.java

```diff
@@ -207,6 +212,31 @@ public String getCreatedAt() {
         return createdAt;
     }
 
+    /**
+     * Get the Message Received Date/Time in ZonedDateTime format
+     * @return the date at which the message was sent/created/received
+     */
+    public ZonedDateTime getCreatedDateTime() {
+        return createdDateTime;
+    }
+
+    public void setCreatedDateTime(ZonedDateTime createdDateTime) {
+        this.createdDateTime = createdDateTime;
+    }
+
+    /**
+     * Get the Message Update Date/Time in  ZonedDateTime format
+     * @return the date on which the message was updated (markAsRead fires the update event)
+     * @see me.shivzee.callbacks.EventListener
+     */
+    public ZonedDateTime getUpdatedDateTime() {
+        return updatedDateTime;
+    }
+
+    public void setUpdatedDateTime(ZonedDateTime updatedDateTime) {

```

Comment:
as mentioned above reason for `setCreatedDateTime`
remove this function `setUpdatedDateTime(ZonedDateTime)`